MSAL/src/public/configuration/global/MSALTelemetryConfig.h

```diff
@@ -0,0 +1,87 @@
+//------------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+#import <Foundation/Foundation.h>
+
+
+/*!
+ @protocol MSALTelemetryDispatcher
+ 
+ Developer should implement it in order to receive telemetry events.
+ 
+ Usage: an instance of MSALTelemetryDispatcher implementation is required when registerring dispatcher for MSALTelemetry.
+ */
+@protocol MSALTelemetryDispatcher <NSObject>
+
+/*!
+ Callback function that will be called by MSAL when telemetry events are flushed.
+ @param events events is represented by an array of dictionary of key-value pair of event property name/value.
+ */
+- (void)dispatchEvent:(nonnull NSArray<NSDictionary<NSString *, NSString *> *> *)events;
+/*!
+ Configuration to dispatch only for failures.
+ */
+- (BOOL)onFailureOnly;
```

Comment:
Why is this one necessary as a method?

Reply:
It shouldn't be needed anymore, as addDispatcher will take care of this.